mesonbuild/interpreter/interpreter.py

```diff
@@ -292,7 +309,7 @@ def __init__(
         self.processed_buildfiles: T.Set[str] = set()
         self.project_args_frozen = False
         self.global_args_frozen = False  # implies self.project_args_frozen
-        self.subprojects: T.Dict[str, SubprojectHolder] = {}
+        self.subprojects: T.Dict[T.Tuple[str, MachineChoice], SubprojectHolder] = {}
```

Comment:
We have `PerMachine` for storing things that are per-machine, we should use that here instead of storing it in the dict.

Reply:
Thanks! Will make this change tomorrow.